internal/repository/edgedevice/edgedevice.go

```diff
@@ -81,3 +84,39 @@ func (r *CRRepository) RemoveFinalizer(ctx context.Context, edgeDevice *v1alpha1
 
 	return nil
 }
+
+func (r *CRRepository) UpdateLabels(ctx context.Context, device *v1alpha1.EdgeDevice, labels map[string]string) error {
+	err := r.updateLabels(ctx, device, labels)
+	if err == nil {
+		return nil
+	}
+
+	// retry patching the edge device labels
```

Comment:
Could you please add a comment why we need to retry here?

Reply:
Sure, will add